server/controllers/index.js

```diff
@@ -69,13 +72,15 @@ export const updateKata = async (req, res) => {
     const { name, slug, link, score } = req.body
 
     const katathon = await Katathon.findById(katathonId)
-    katathon.katas.map(kata => {
-      if(kata.kataId === kataId) {
+
+    katathon.katas = katathon.katas.map(kata => {
+      if(kata.kataId == kataId) {
         if(name) kata.name = name
```

Comment:
Can we use `Object.assign()` here to merge available properties in to the kata object?
(You will need to create a new object from the `req.body` with the destructured properties first)
Then you can transform the `if` into a ternary:
`return (kata.kataId == kataId) ? Object.assign({}, kata, bodyCopy) : kata`